src/document.c

```diff
@@ -1264,40 +1266,14 @@ void document_apply_indent_settings(GeanyDocument *doc)
 
 void document_show_tab(GeanyDocument *doc)
 {
-	if (show_tab_idle)
-	{
-		g_source_remove(show_tab_idle);
-		show_tab_idle = 0;
-	}
-
 	gtk_notebook_set_current_page(GTK_NOTEBOOK(main_widgets.notebook),
 		document_get_notebook_page(doc));
 
 	/* finally, let the editor widget grab the focus so you can start coding
 	 * right away */
-	document_try_focus(doc, NULL);
-}
-
-
-static gboolean show_tab_cb(gpointer data)
-{
-	GeanyDocument *doc = (GeanyDocument *) data;
-
-	show_tab_idle = 0;
-	/* doc might not be valid e.g. if user closed a tab whilst Geany is opening files */
-	if (DOC_VALID(doc))
-		document_show_tab(doc);
-
-	return G_SOURCE_REMOVE;
-}
-
-
-void document_show_tab_idle(GeanyDocument *doc)
-{
-	if (show_tab_idle)
-		g_source_remove(show_tab_idle);
-
-	show_tab_idle = g_idle_add(show_tab_cb, doc);
+	/* FIXME: is that actually useful??  I don't see any difference disabling this code... */
+	if (!main_status.opening_session_files)
+		document_try_focus(doc, NULL);
```

Comment:
This I don't know, I couldn't find a case where it was needed, but I guess there was a reason for it…